docs/exposing_dns.md

```diff
@@ -0,0 +1,63 @@
+# Exposing DNS UDP traffic for k8gb
+
+In order for k8gb to function properly, associated CoreDNS service deployed with k8gb needs be exposed for external DNS UDP traffic on cluster worker nodes.
+
+Actual ways to achieve this depend on many factors, such as underlying infrastructure (cloud, on-prem, managed vs bare-metal setup), means to expose CoreDNS service (ClusterIP, LoadBalancer),
+type of load balancer or ingress controller used etc.
+This topic lies outside of k8gb project  scope, as often the related configuration is shared by cluster services, requires additional permissions, and as result can't be owned by k8gb controller deployment.
+However, we can touch few examples with popular configurations in k8s, also ones we have thoroughly tested in local setup and production environments.
+
+## Ingress Controller with UDP support (NGINX)
+
+> *Check [NGINX Ingress controller official documentation](https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/) for additional information*
+
+Ingress resource doesn't support TCP or UDP services in general. In order to let NGINX Ingress controller know that we want to expose UDP port for k8gb CoreDNS service (`k8gb-coredns`), we need to create or patch `udp-services` ConfigMap in a  namespace where NGINX ingress controller is installed (`ingress-nginx` by default).
+This can be done by applying the following manifest:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: udp-services
+  namespace: ingress-nginx
+data:
+  53: "k8gb/k8gb-coredns:53"
+```
+or by running `kubectl` one-liner:
+```sh
+# Patch the existing `udp-services` ConfigMap in NGINX Ingress controller namespace:
+
+kubectl patch -n ingress-nginx -p '{"data":{"53":"k8gb/k8gb-coredns:53"}}' --type=merge cm/udp-services
+
+# Or create `udp-services` ConfigMap if it doesn't exist, e.g.:
+
+kubectl create -n ingress-nginx cm udp-services --from-literal="53"="k8gb/k8gb-coredns:53"
+```
+
+[Local project setup](./local.md) does this patching automatically.
+
+## External load balancer
+
+CoreDNS can be also exposed for DNS UDP traffic via external load balancer,
+if underlying infrastructure supports that.<br>
+[AWS EKS](https://aws.amazon.com/eks) with [NLB](https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html) and [k3d](https://www.k3d.io) with [ServiceLB](https://rancher.com/docs/k3s/latest/en/networking/#service-load-balancer) are good examples of such an infrastructure proven to work for k8gb deployments.
+
+In general, `Service` resource configuration for k8gb CoreDNS in this case looks like:
```

Comment:
Probably it worth mentioning that we are using this approach in our AWS+Route53 reference setup at https://github.com/AbsaOSS/k8gb/blob/master/docs/deploy_route53.md and that the Service resource is created by k8gb helm chart. Otherwise, it creates the impression of manual Service primitive configuration

Reply:
Amended.